qualtran/surface_code/azure_cost_model.ipynb

```diff
@@ -0,0 +1,708 @@
+    "$$\n",
+    "a \\log_2{\\frac{1}{\\epsilon}} + b\n",
+    "$$\n",
+    "\n",
+    "Where $a$ and $b$ are constants that depend on the gate set and the approximation protocol. Table 1 of [Kliuchnikov et al](https://arxiv.org/abs/2203.10064) gives estimates for those constants for different combinations of gate sets and protocols.\n",
+    "\n",
+    "In [Beverland et al](https://arxiv.org/abs/2211.07629), Although they don't specify the approximation algorithm their constants of $a=0.53, b=5.3$ are closest to the mixed fallback approximation protocol with a Clifford+T gate set constants of $a=0.53, b=4.86$. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from qualtran.surface_code.rotation_cost_model import BeverlandEtAlRotationCost\n",
+    "\n",
+    "BeverlandEtAlRotationCost"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Quantum Error Correction Scheme\n",
+    "The quantum error correction scheme determines three things:\n",
+    "1. The logical error rate given a code distance and physical error rate $P(d)$.\n",
+    "1. The number of physical qubits needed per logical qubit $n(d)$.\n",
+    "1. The number of logical time steps $\\tau(d)$.\n",
+    "\n",
+    "Tabel V of [Beverland et al](https://arxiv.org/abs/2211.07629) lists how these are computed from the QEC scheme.\n",
+    "\n",
+    "In [Beverland et al](https://arxiv.org/abs/2211.07629), they use a gate-based QEC that has $P(d), n(d), \\tau(d)$ as:\n",
+    "$$\n",
+    "P(d) = 0.03 \\left ( \\frac{p}{0.01} \\right) ^ \\frac{d+1}{2}\\\\\n",
+    "n(d) = 2 d^2\\\\\n",
+    "\\tau(d) = \\textrm{\\{error detection circuit time\\}} \\cdot d\\\\\n",
+    "$$\n",
+    "\n",
+    "The error detection circuit time depends on several factors including the time to apply a Clifford, measurement and reset operations as well as classical processing.\n",
+    "\n",
+    "In [Beverland et al](https://arxiv.org/abs/2211.07629) Table V they don't take into account the classical processing part and assume that a reset takes the same time as a measurement leading to the formula:\n",
+    "$$\n",
+    "\\textrm{\\{error detection circuit time\\}} = 4t_\\textrm{gate} + 2t_\\textrm{meas}\n",
+    "$$\n",
+    "\n",
+    "Other authors (e.g. [Fowler, Gidney](https://arxiv.org/abs/1808.06709)) assume that the entire process takes a specific time (e.g. $\\approx 1\\mu s$).\n",
+    "\n",
+    "In Qualtran, we use \"error detection circuit time\" rather than fall to the low level hardware parameters. "
```

Comment:
"error detection circuit time" isn't what the parameter is actually called. These should be consistent (or consistent-ish).

I like that you contrast the two ways of putting in the physical assumptions. It would be nice for this final sentence to say which one Qualtran follows. If I understand correctly, we're using a Gidney Fowler parameter even though it has this decomposition in the azure model

Reply:
rewritten